modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py

```diff
@@ -383,37 +394,53 @@ def test_connection_to_extended_node(extended_node, stop_on_failure, upgrade_sta
     return True
 
 
-def run_preflight_validations(stop_on_failure, target_version, agentless_gw_extended_node_dict,
-                              dsf_hub_extended_node_dict, upgrade_status_service):
+def run_preflight_validations(
+        stop_on_failure, target_version, ignore_healthcheck_warning,
+        ignore_healthcheck_checks, agentless_gw_extended_node_dict,
+        dsf_hub_extended_node_dict, upgrade_status_service):
     print("----- Preflight validations")
 
     successful = True
     for extended_node in chain(agentless_gw_extended_node_dict.values(), dsf_hub_extended_node_dict.values()):
         if upgrade_status_service.should_run_preflight_validations(extended_node.get('dsf_node_id')):
-            node_successful = run_preflight_validations_for_node(stop_on_failure, target_version, extended_node, upgrade_status_service)
+            node_successful = run_preflight_validations_for_node(
+                stop_on_failure, target_version,
+                ignore_healthcheck_warning, ignore_healthcheck_checks,
+                extended_node, upgrade_status_service)
             successful = successful and node_successful
 
     return successful
 
 
-def run_preflight_validations_for_node(stop_on_failure, target_version, extended_node, upgrade_status_service):
+def run_preflight_validations_for_node(
+        stop_on_failure, target_version,
+        ignore_healthcheck_warning, ignore_healthcheck_checks,
+        extended_node, upgrade_status_service):
     error_message = None
     try:
         upgrade_status_service.update_upgrade_status(
             extended_node.get('dsf_node_id'), UpgradeStatus.RUNNING_PREFLIGHT_VALIDATIONS,
         )
-        preflight_validations_result = run_preflight_validations_script(
-            target_version, extended_node.get('dsf_node'), extended_node.get('dsf_node_name'), extended_node.get('python_location'),
-        )
-
-        if are_preflight_validations_passed(preflight_validations_result):
-            print(f"### Preflight validations passed for {extended_node.get('dsf_node_name')}")
-            upgrade_status_service.update_upgrade_status(
-                extended_node.get('dsf_node_id'), UpgradeStatus.PREFLIGHT_VALIDATIONS_SUCCEEDED,
-            )
+        version_list = extended_node['sysconfig']['JSONAR_VERSION'].split('-')[0].split('.')
+        version_tuple = tuple(map(int, version_list))
+        if version_tuple >= (4, 15):
+            script_output = run_remote_script(extended_node['dsf_node'], "", f"sudo {extended_node['sysconfig']['JSONAR_BASEDIR']}/bin/health-checker --quiet --output JSON --target-version {target_version}")
```

Comment:
Extract this code to function run_healthcheck because the parent function is now too long and the extraction will make the code more readable. (P0)